src/components/PostDetail.client.jsx

```diff
@@ -0,0 +1,55 @@
+const createPostFetcher = function (permalink) {
+  return {
+    data: null,
+    promise: null,
+    fetch(id) {
+      if (this.data != null) {
+        return this.data;
+      }
+
+      if (this.promise == null) {
+        const payload = {
+          component: "PostDetail",
+          props: {
+            permalink,
+          },
+        };
+
+        this.promise = fetch("/render", {
+          body: JSON.stringify(payload),
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        })
+          .then((res) => res.text())
+          .then((str) => {
+            this.data = JSON.parse(str, (key, value) => {
+              if (key === "$$typeof") {
+                if (value === "Symbol(react.element)") {
```

Comment:
symbol is not serializable, we replace it with string on server and revive it back on client